### PR TITLE
#11 Statistical analysis tools

### LINKED ISSUE #11: Statistical analysis tools
i want to add phpstan and psalm what you think?

### PR DESCRIPTION
**Pull request by tigokr**
- phpstan 8 level

### CODE REVIEW

File: src/Aggregation/Reduced.php [hunk 1]
```diff
@@ -42,7 +46,7 @@ public function __construct(Arrayee $arrayee, callable $callback, Arrayee $initi
     /**
      * @param Arrayee $arrayee
      * @param callable $callback Function that reduces items to array <code>function(array $carry, $item): array</code>
-     * @return Reduced
+     * @return Reduced<V>
```
Comment (stepanets) [inline src/Aggregation/Reduced.php:49]: is it somehow supported by phpstorm?

Comment (tigokr) [reply src/Aggregation/Reduced.php:49]: yep, below you can see a warning if I remove a template definition from Reduced class
image

Also, [here](https://blog.jetbrains.com/phpstorm/2020/12/phpstorm-2020-3-release/) described more detailed (see Psalm and PHPStan Support section)

**Comment by stepanets** [comment]
can we do something  with deprecation warnings?

https://travis-ci.com/github/elegant-bro/arrayee/jobs/468863083

**Comment by tigokr** [comment]
I'm afraid it's impossible to avoid these [deprecation](https://www.php.net/manual/en/reflectiontype.tostring.php) warnings because our support of PHP 7.1 requires to use an old version of `symplify/easy-coding-standard` -> `symplify/package-builder` https://github.com/symplify/package-builder/blob/fbdfe363a27070cfdfbc47d5f59e711ed08bb060/src/DependencyInjection/CompilerPass/AutoReturnFactoryCompilerPass.php#L131

only skipping of 7.1 support will recover these errors

**Comment by stepanets** [comment]
> 
> 
> I'm afraid it's impossible to avoid these [deprecation](https://www.php.net/manual/en/reflectiontype.tostring.php) warnings because our support of PHP 7.1 requires to use an old version of `symplify/easy-coding-standard`
> 
> only skipping of 7.1 support will recover these errors

ok, we think about it later)

**Review by stepanets — APPROVED** [review]

**Pull request merged**